DigitalLearningSolutions.Data/Services/ActionPlanService.cs

```diff
@@ -68,10 +82,43 @@ public void AddResourceToActionPlan(int competencyLearningResourceId, int delega
 
             foreach (var competencyId in learningLogCompetenciesToAdd)
             {
-                learningLogItemsDataService.InsertLearningLogItemCompetencies(learningLogItemId, competencyId, addedDate);
+                learningLogItemsDataService.InsertLearningLogItemCompetencies(
+                    learningLogItemId,
+                    competencyId,
+                    addedDate
+                );
             }
 
             transaction.Complete();
         }
+
+        public async Task<IEnumerable<ActionPlanItem>> GetIncompleteActionPlanItems(int delegateId)
+        {
+            if (!config.IsSignpostingUsed())
+            {
+                return new List<ActionPlanItem>();
+            }
+
+            var incompleteLearningLogItems = learningLogItemsDataService.GetLearningLogItems(delegateId)
+                .Where(
+                    i => i.CompletedDate == null && i.ArchivedDate == null && i.LearningHubResourceReferenceId != null
+                ).ToList();
+
+            if (!incompleteLearningLogItems.Any())
+            {
+                return new List<ActionPlanItem>();
+            }
+
+            var incompleteResourceIds = incompleteLearningLogItems.Select(i => i.LearningHubResourceReferenceId!.Value);
+            var bulkResponse = await learningHubApiClient.GetBulkResourcesByReferenceIds(incompleteResourceIds);
+            var matchedLearningResources = bulkResponse.ResourceReferences;
+            var incompleteActionPlanItems = matchedLearningResources.Select(

```

Comment:
I don't really think it's worth assigning this here, just use it on the next line.